confgenerator/jvm.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package confgenerator
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/otel"
+	"github.com/kardianos/osext"
+)
+
+type MetricsReceiverJVM struct {
+	ConfigComponent `yaml:",inline"`
+
+	MetricsReceiverShared `yaml:",inline"`
+
+	Endpoint string `yaml:"endpoint" validate:"omitempty,url"`
+}
+
+const defaultJVMEndpoint = "localhost:9999"
+
+func (r MetricsReceiverJVM) Type() string {
+	return "jvm"
+}
+
+func (r MetricsReceiverJVM) Pipelines() []otel.Pipeline {
+	if r.Endpoint == "" {
+		r.Endpoint = defaultJVMEndpoint
+	}
+
+	jarPath, err := findJarPath()
+	if err != nil {
+		_ = err // TODO(djaglowski) return nil, fmt.Errorf("create jvmreceiver: %v, err)
+	}
+
+	return []otel.Pipeline{{
+		Receiver: otel.Component{
+			Type: "jmx",
+			Config: map[string]interface{}{
```

Comment:
Mentioned this to Craig in standup today, we need this to include credentials (username/password) for JMX.